specs/NotifyDropTargetAction.md

```diff
@@ -0,0 +1,223 @@
+  /// This function corresponds to IDropTarget::DragEnter
+  ///
+  /// The hosting application must register as an IDropTarget and implement
+  /// and forward DragEnter calls to this function.
+  ///
+  /// In addition, the hosting application needs to create an IDropTargetHelper
+  /// and call the corresponding IDropTargetHelper::DragEnter function on that
+  /// object before forwarding the call to WebView.
+  ///
+  /// point parameter must be modified to include the WebView's offset and be in
+  /// the WebView's client coordinates (Similar to how SendMouseInput works).
+  HRESULT DragEnter(
+      [in] IDataObject* dataObject,
+      [in] DWORD keyState,
+      [in] POINT point,
+      [out, retval] DWORD* effect);
+
+  /// This function corresponds to IDropTarget::DragLeave
+  ///
+  /// The hosting application must register as an IDropTarget and implement
+  /// and forward DragLeave calls to this function.
+  ///
+  /// In addition, the hosting application needs to create an IDropTargetHelper
+  /// and call the corresponding IDropTargetHelper::DragLeave function on that
+  /// object before forwarding the call to WebView.
+  HRESULT DragLeave();
+
+  /// This function corresponds to IDropTarget::DragOver
+  ///
+  /// The hosting application must register as an IDropTarget and implement
+  /// and forward DragOver calls to this function.
+  ///
+  /// In addition, the hosting application needs to create an IDropTargetHelper
+  /// and call the corresponding IDropTargetHelper::DragOver function on that
+  /// object before forwarding the call to WebView.
+  ///
+  /// point parameter must be modified to include the WebView's offset and be in
+  /// the WebView's client coordinates (Similar to how SendMouseInput works).
+  HRESULT DragOver(
+      [in] DWORD keyState,
+      [in] POINT point,
+      [out, retval] DWORD* effect);
+
+  /// This function corresponds to IDropTarget::Drop
+  ///
+  /// The hosting application must register as an IDropTarget and implement
+  /// and forward Drop calls to this function.
+  ///
+  /// In addition, the hosting application needs to create an IDropTargetHelper
+  /// and call the corresponding IDropTargetHelper::Drop function on that
+  /// object before forwarding the call to WebView.
+  ///
+  /// point parameter must be modified to include the WebView's offset and be in
+  /// the WebView's client coordinates (Similar to how SendMouseInput works).
+  HRESULT Drop(
+      [in] IDataObject* dataObject,
+      [in] DWORD keyState,
+      [in] POINT point,
+      [out, retval] DWORD* effect);
+}
+```
+## WinRT
+```c++
+namespace Microsoft.Web.WebView2.Core
+{
+  uint32_t DragEnter(
+      winrt.Windows.ApplicationModel.DataTransfer.DataPackage const& dataObject,
+      uint32_t keyState,
+      winrt.Windows.Foundation.Point point);
+
+  void DragLeave();
+
+  uint32_t DragOver(
+      uint32_t keyState,
+      winrt::Windows.Foundation.Point point);
+
+  uint32_t Drop(
```

Comment:
The winrt APIs are just like the COM APIs - which is fine if this API is only for win32 style drag and drop. If you have a UWP app with a CoreWindow and such, do these APIs fit into that? If not and this is only for win32 style drag and drop just need to make that explicit somewhere that we have winrt APIs for this but would only make sense for winrt in a win32 app.

Reply:
I think we discussed this in e-mail, but it should work when we do visual hosting in an app written in winrt in win32 or uwp context.